src/qibo/transpiler/optimizer.py

```diff
@@ -66,3 +67,261 @@ def __call__(self, circuit: Circuit):
             else:
                 new.add(fgate)
         return new
+
+
+class InverseCancellation(Optimizer):
+    """
+    Identifies and removes pairs of adjacent gates from a quantum circuit.
+    """
+
+    def __init__(self):
```

Comment:
Move all class docstrings to their `__init__` methods and add `args` when necessary.